### PR TITLE
Upgrade to rubocop 1.61 and rubocop-rspec 3.0.3

### PR DESCRIPTION
**Pull request by tecden**

### CODE REVIEW

File: config/ext/rspec.yml [hunk 1]
```diff
@@ -278,9 +279,10 @@ RSpec/DescribedClass:
   SupportedStyles:
     - described_class
     - explicit
+  OnlyStaticConstants: false
```
Comment (tecden) [inline config/ext/rspec.yml:282]: Habe nicht ganz verstanden worum es hier geht. 
https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#onlystaticconstants-true-default

File: config/ext/rspec.yml [hunk 2]
```diff
@@ -356,10 +359,17 @@ RSpec/EmptyLineAfterSubject:
 
 RSpec/EmptyMetadata:
   Description: Avoid empty metadata hash.
-  Enabled: pending
+  Enabled: true
   VersionAdded: '2.24'
+  VersionChanged: '2.31'
   Reference: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/EmptyMetadata
 
+RSpec/EmptyOutput:
```
Comment (tecden) [inline config/ext/rspec.yml:367]: Fand ich gut :+1: 
https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecemptyoutput

File: config/ext/rspec.yml [hunk 3]
```diff
@@ -435,32 +445,23 @@ RSpec/ExpectInHook:
   VersionAdded: '1.16'
   Reference: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ExpectInHook
 
+RSpec/ExpectInLet:
```
Comment (tecden) [inline config/ext/rspec.yml:448]: https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecexpectinlet

:+1:

File: config/ext/rspec.yml [hunk 4]
```diff
@@ -435,32 +445,23 @@ RSpec/ExpectInHook:
...
+  VersionAdded: '2.30'
+  Reference: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ExpectInLet
+
 RSpec/ExpectOutput:
   Description: Checks for opportunities to use `expect { ... }.to output`.
   Enabled: true
   VersionAdded: '1.10'
   Reference: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ExpectOutput
 
-RSpec/FilePath:
```
Comment (tecden) [inline config/ext/rspec.yml:444]: Wurde entfernt in 3.0
https://github.com/rubocop/rubocop-rspec/releases/tag/v3.0.0

File: config/ext/rspec.yml [hunk 5]
```diff
@@ -543,6 +545,13 @@ RSpec/InstanceVariable:
   StyleGuide: https://rspec.rubystyle.guide/#instance-variables
   Reference: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/InstanceVariable
 
+RSpec/IsExpectedSpecify:
```
Comment (tecden) [inline config/ext/rspec.yml:548]: Hab hier keine große Meinung. Hab ich aber auch noch nie bei uns gesehen.
https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecexpectinlet :+1:

File: config/ext/rspec.yml [hunk 6]
```diff
@@ -636,6 +645,12 @@ RSpec/MissingExampleGroupArgument:
   VersionAdded: '1.28'
   Reference: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/MissingExampleGroupArgument
 
+RSpec/MissingExpectationTargetMethod:
```
Comment (tecden) [inline config/ext/rspec.yml:648]: https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecmissingexpectationtargetmethod :+1:

File: config/ext/rspec.yml [hunk 7]
```diff
@@ -762,10 +777,22 @@ RSpec/ReceiveNever:
 
 RSpec/RedundantAround:
   Description: Remove redundant `around` hook.
-  Enabled: pending
+  Enabled: true
   VersionAdded: '2.19'
   Reference: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/RedundantAround
 
+RSpec/RedundantPredicateMatcher:
```
Comment (tecden) [inline config/ext/rspec.yml:784]: [RedundantPredicateMatcher ](https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecredundantpredicatematcher) :+1:

File: config/ext/rspec.yml [hunk 8]
```diff
@@ -762,10 +777,22 @@ RSpec/ReceiveNever:
...
+  Description: Checks for redundant predicate matcher.
+  Enabled: true
+  VersionAdded: '2.26'
+  Reference: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/RedundantPredicateMatcher
+
+RSpec/RemoveConst:
```
Comment (tecden) [inline config/ext/rspec.yml:790]: https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecredundantpredicatematcher

Hab ich noch nie gesehen. Mir fällt nicht ein, wozu man das brauchen könnte `remove_const` unbedingt im Test braucht. Von mir aus :+1:

File: config/ext/rspec.yml [hunk 9]
```diff
@@ -796,6 +823,12 @@ RSpec/RepeatedIncludeExample:
   VersionAdded: '1.44'
   Reference: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/RepeatedIncludeExample
 
+RSpec/RepeatedSubjectCall:
```
Comment (tecden) [inline config/ext/rspec.yml:826]: [RepeatedSubjectCal](https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecrepeatedsubjectcall) :+1:

File: config/ext/rspec.yml [hunk 10]
```diff
@@ -896,6 +935,12 @@ RSpec/SubjectStub:
   StyleGuide: https://rspec.rubystyle.guide/#dont-stub-subject
   Reference: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/SubjectStub
 
+RSpec/UndescriptiveLiteralsDescription:
```
Comment (tecden) [inline config/ext/rspec.yml:938]: https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecundescriptiveliteralsdescription

Sollte imho dem Entwickler überlassen sein :-1:

File: config/ext/rspec.yml (hunk 1 above)
Comment (brunosedler) [reply config/ext/rspec.yml:282]: I'm not 100% sure either but I to my understading, with this setting set to false, you cannot write
```
describe MyClass do
  subject { MyClass::CONSTANT }
end
```

but you have to write
```
describe MyClass do
  subject { described_class::CONSTANT }
end
```
instead

File: config/ext/rspec.yml (hunk 2 above)
Comment (brunosedler) [reply config/ext/rspec.yml:367]: I don't think I have ever needed this, so I'm not sure why I would test for empty output. However, I think I'd prefer disabling it, because when I read  `expect { bar }.to output.to_stderr` I would not expect, that specifically an empty string is meant, but rather that **anything** should have been written to stderr. 
When you compare this for example to the `raise_error` matcher, when you see `expect { do_something }.to raise_error`, it means any error is raised, not one with an empty message. 

TLDR: I'd prefer to more specific, because I think omissions can lead to confusion here.

File: config/ext/rspec.yml (hunk 5 above)
Comment (brunosedler) [reply config/ext/rspec.yml:548]: I think we usually just use `it` in all cases so I don't have a strong opinion here either.

File: config/ext/rspec.yml (hunk 6 above)
Comment (brunosedler) [reply config/ext/rspec.yml:648]: I assume for `should`-syntax, this Cop is ignored, right? Because otherwise it would mark all cases where you still use `should`

File: config/ext/rspec.yml (hunk 8 above)
Comment (brunosedler) [reply config/ext/rspec.yml:790]: I don't think I've seen this either, so fine by me.

File: config/ext/rspec.yml (hunk 10 above)
Comment (brunosedler) [reply config/ext/rspec.yml:938]: Agree. Please set enabled to false

**Review by brunosedler — COMMENTED** [review]
I have a different opinion on one or two cops. Maybe we need @codener as a tie breaker on those

File: config/ext/rspec.yml (hunk 1 above)
Comment (codener) [reply config/ext/rspec.yml:282]: I think we've had a discussion on this somewhere. Let's not enforce this, thus disable the cop.

File: config/ext/rspec.yml (hunk 2 above)
Comment (codener) [reply config/ext/rspec.yml:367]: This cop is a detail, but reasonable in my eyes and it should not generate noise. Voting for enabling.

> when I read expect { bar }.to output.to_stderr I would not expect, that specifically an empty string is meant, but rather that anything should have been written to stderr.
When you compare this for example to the raise_error matcher, when you see expect { do_something }.to raise_error, it means any error is raised, not one with an empty message.

I believe this is a misunderstanding. The cop states that "does not print '' to stderr" is equivalent to "prints anything to stderr", which is fine IMO.

File: config/ext/rspec.yml (hunk 5 above)
Comment (codener) [reply config/ext/rspec.yml:548]: I would never expect a `let` to contain expectations. Voting for enabling.

**Comment by codener** [comment]
I have added my 50c.
